lib/Image/Gotenberg.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Image;
+
+use Gotenberg\Gotenberg as GotenbergAPI;
+use Pimcore\Config;
+use Pimcore\Logger;
+
+/**
+ * @internal
+ */
+class Gotenberg
+{
+    public static function isSupported(): bool
+    {
+        return class_exists(GotenbergAPI::class);
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public static function convert(string $url, string $outputFile, ?string $sessionName = null, ?string $sessionId = null, string $windowSize = '1280,1024'): bool
+    {
+        $outputPath = dirname($outputFile);
+        $filename = basename($outputFile, '.png');
+
+        try {
+            $headers = [];
+            if (php_sapi_name() !== 'cli') {
+                if (null !== $sessionId && null !== $sessionName) {
+                    $headers['Cookie'] = $sessionName . '=' . $sessionId;
+                }
+            }
+
+            $gotenbergBaseUrl = Config::getSystemConfiguration('gotenberg')['base_url'];
+            $chromium = GotenbergAPI::chromium($gotenbergBaseUrl);
+
+            if (!empty($headers)) {
+                $chromium->extraHttpHeaders($headers);
+            }
+
+            list($paperWidth, $paperHeight) = explode(',', $windowSize);
+
+            // the PDF paper size works in inches, 96 is to convert pixels to inches
+            $chromium->paperSize((int)$paperWidth / 96, (int)$paperHeight / 96);
+            $chromium->printBackground();
+            $chromium->emulateScreenMediaType();
+            $chromium->nativePageRanges('1');
+
+            $request = $chromium->outputFilename($filename)->url($url);
+            GotenbergAPI::save($request, $outputPath);
```

Comment:
This generated a PDF right? 🤔

Reply:
yes and Imagick [converts ](https://www.php.net/manual/de/class.imagick.php#111197)it to an image [here](https://github.com/pimcore/pimcore/pull/14212/files?w=1#diff-47329e5b5e592052262d0eb596ba4650c40b1d3286a990c54efde4e7fd64b7a6R1161-R1163)

Reply:
But this is only for version comparisons, this is also used to generate preview images here: 
https://github.com/pimcore/pimcore/blob/8d9b9cf0fd1ba65ccf7485eb0be0547493ff267f/models/Document/Service.php#L604

Where it could be that `Imagick` is not available, but GD2 is used. 
And to be honest I don't like the idea of haven PDF in-between, probably better than to stay on Chromium for doing the HTML 2 PNG/Image stuff. WDYT?

Reply:
yes, i am with you, let's put this aside and reconsider when this [PR](https://github.com/gotenberg/gotenberg/pull/504) gets merged and release.